core/common/src/main/java/alluxio/security/authentication/DefaultAuthenticationServer.java

```diff
@@ -0,0 +1,230 @@
+    implements AuthenticationServer {
+  private static final Logger LOG = LoggerFactory.getLogger(DefaultAuthenticationServer.class);
+
+  /** List of channels authenticated against this server. */
+  protected final ConcurrentHashMap<UUID, AuthenticatedChannelInfo> mChannels;
+  /** Scheduler for periodic cleaning of channels registry. */
+  protected final ScheduledExecutorService mScheduler;
+
+  /** Interval for clean-up task to fire. */
+  protected final long mCleanupIntervalMs =
+      Configuration.getMs(PropertyKey.AUTHENTICATION_INACTIVE_CHANNEL_REAUTHENTICATE_PERIOD);
+
+  /**
+   * Creates {@link DefaultAuthenticationServer} instance.
+   */
+  public DefaultAuthenticationServer() {
+    checkSupported(Configuration.getEnum(PropertyKey.SECURITY_AUTHENTICATION_TYPE, AuthType.class));
+    mChannels = new ConcurrentHashMap<>();
+    mScheduler = Executors.newScheduledThreadPool(1,
+        ThreadFactoryUtils.build("auth-cleanup", true));
+    mScheduler.scheduleAtFixedRate(() -> {
+      cleanupStaleClients();
+    }, mCleanupIntervalMs, mCleanupIntervalMs, TimeUnit.MILLISECONDS);
+  }
+
+  @Override
+  public StreamObserver<SaslMessage> authenticate(StreamObserver<SaslMessage> responseObserver) {
+    // Create and return server sasl driver that will coordinate authentication traffic.
+    SaslStreamServerDriver driver = new SaslStreamServerDriver(this);
+    driver.setClientObserver(responseObserver);
+    return driver;
+  }
+
+  @Override
+  public void registerChannel(UUID channelId, String authorizedUser, SaslServer saslServer) {
+    if (null != mChannels.putIfAbsent(channelId,
+        new AuthenticatedChannelInfo(authorizedUser, saslServer))) {
+      mChannels.remove(channelId);
+      throw new RuntimeException(String
+          .format("Channel: %s already exists in authentication registry.", channelId.toString()));
+    }
+    LOG.debug("Registered new channel:" + channelId);
+  }
+
+  @Override
+  public String getUserNameForChannel(UUID channelId) throws UnauthenticatedException {
+    if (mChannels.containsKey(channelId)) {
+      AuthenticatedChannelInfo clientInfo = mChannels.get(channelId);
+      return clientInfo.getUserName();
+    } else {
+      throw new UnauthenticatedException(
+          String.format("Client:%s needs to be authenticated", channelId.toString()));
+    }
+  }
+
+  @Override
+  public void unregisterChannel(UUID channelId) {
+    AuthenticatedChannelInfo channelInfo = mChannels.remove(channelId);
+    if (channelInfo != null) {
+      try {
+        channelInfo.getSaslServer().dispose();
+      } catch (SaslException e) {
+        LOG.warn("Failed to dispose sasl client for channel-Id: {}. Error: {}", channelId,
+            e.getMessage());
+      }
+    }
+  }
+
+  /**
+   * Primitive that is invoked periodically for cleaning the registry from clients that has become
+   * stale.
+   */
+  private void cleanupStaleClients() {
+    LocalTime cleanupTime = LocalTime.now();
```

Comment:
Is local time slower than `System.currentTimeMillis()`?

Reply:
I don't understand the question. The reason I used LocalTime was that it has nice tools like addSeconds, isBefore,...